### PR TITLE
docs: add tutorial + introduction for full mev emulation with eth2-package

### PR DESCRIPTION
**Pull request by leeederek**
## Description:
Adds a tutorial for spinning up the eth2-package with full MEV.

Includes:
- a section talking about what and why MEV emulation
- a section (briefly) reviewing the architecture
- the tutorial/guide itself 

An argument can probably be made to keep this strictly tutorial-only, but i believe the extra context is valuable.

## Is this change user facing?
YES, absolutely

### CODE REVIEW

File: docs/docs/guides/how-to-full-mev-with-eth2-package.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+Keep reading to learn [how it all works](#brief-overview-of-the-architecture) & [how to get started with the `eth2-package`](#quickstart).
+
+#### Why `eth2-package`?
+As a reminder, the [`eth2-package`](https://github.com/kurtosis-tech/eth2-package) is a reproducible and portable environment definition that should be used to bootstrap & deploy private testnets. The package will function the exact same way locally or in the cloud over Docker or Kubernetes, supports all major Execution Layer (EL) and Consensus Layer (CL) client implementations, and can be scaled to whatever size your team needs - limited only by your underlying hardware/backend.
+
+#### What if I only want the MEV parts?
+And if that wasn't enough, Kurtosis environment definitions (known as [Packages](https://docs.kurtosis.com/concepts-reference/packages/)) are entirely composable, meaning you can define and build-your-own private testnet using only the parts you need and with the option of adding your own services (e.g. MEV searcher tools). Feel free to check out [eth-kurtosis](https://github.com/kurtosis-tech/eth-kurtosis) for how to do this!
```
Comment (h4ck3rk3y) [inline docs/docs/guides/how-to-full-mev-with-eth2-package.md:25]: do we link out to this still?

File: docs/docs/guides/how-to-full-mev-with-eth2-package.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+The `eth2-package` with MEV emulation are already in use by client teams to help uncover bugs (examples [here](https://github.com/prysmaticlabs/prysm/pull/12736) and [here](https://github.com/NethermindEth/nethermind/commit/4d805769159dc0717aa1ba38cc3ebc53f9a375cf)). 
+
+Everything you see below in the architecture diagram gets configured, initialized, and bootstrapped together by Kurtosis.
+
+![mev-arch](../../static/img/guides/full-mev-infra-arch-diagram.png)
+
+#### Caveats:
+* The `mev-boost` service (a sidecar for the consensus layer client) requires Capella at an epoch of non-zero. For the eth2-package, the Capella fork is set to happen after the first epoch to be started up and fully connected to the CL client.
```
Comment (h4ck3rk3y) [inline docs/docs/guides/how-to-full-mev-with-eth2-package.md]: mev-boost-relay requires this!!

File: docs/docs/guides/how-to-full-mev-with-eth2-package.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+:::note
+Quick aside on what `mev-flood` does:
+Once the network is online, `mev-flood` will deploy UniV2 smart contracts, provision liquidity on UniV2 pairs, & begin to send a constant stream of UniV2 swap transactions to the network's public mempool. Depending on the mode you're running, either the `mock-builder` or Flashbot's `mev-builder`, the transactions will be bundled into payloads for downstream use by the relayer or by validators themselves. It is important to note that `mev-flood` will only be initialized with the `full-mev` set up and will send transactions with a block value of 0. Read more about [`mev-flood` here](https://github.com/flashbots/mev-flood).
```
Comment (h4ck3rk3y) [inline docs/docs/guides/how-to-full-mev-with-eth2-package.md]: mev-flood will send transactions with a non zero block value!!

Comment (h4ck3rk3y) [reply docs/docs/guides/how-to-full-mev-with-eth2-package.md]: non zero block value -> coinbase address is getting richer

**Review by h4ck3rk3y — APPROVED** [review]

File: docs/docs/guides/how-to-full-mev-with-eth2-package.md (hunk 1 above)
Comment (leeederek) [reply docs/docs/guides/how-to-full-mev-with-eth2-package.md:25]: yes

File: docs/docs/guides/how-to-full-mev-with-eth2-package.md (hunk 2 above)
Comment (leeederek) [inline docs/docs/guides/how-to-full-mev-with-eth2-package.md]:
```suggestion
* The `mev-boost-relay` service (a sidecar for the consensus layer client) requires Capella at an epoch of non-zero. For the eth2-package, the Capella fork is set to happen after the first epoch to be started up and fully connected to the CL client.
```

File: docs/docs/guides/how-to-full-mev-with-eth2-package.md (hunk 3 above)
Comment (leeederek) [inline docs/docs/guides/how-to-full-mev-with-eth2-package.md]:
```suggestion
Once the network is online, `mev-flood` will deploy UniV2 smart contracts, provision liquidity on UniV2 pairs, & begin to send a constant stream of UniV2 swap transactions to the network's public mempool. Depending on the mode you're running, either the `mock-builder` or Flashbot's `mev-builder`, the transactions will be bundled into payloads for downstream use by the relayer or by validators themselves. It is important to note that `mev-flood` will only be initialized with the `full-mev` set up and will send transactions with a non-zero block value. Read more about [`mev-flood` here](https://github.com/flashbots/mev-flood). 
```

**Pull request merged**